source/helpers/conditionParser.ts

```diff
@@ -0,0 +1,118 @@
+import { Question, SummaryItem } from '../types/FormTypes';
+
+const evaluateSummaryList = (
+  answers: Record<string, any>,
+  items: SummaryItem[]
+): boolean => {
+  if (!Array.isArray(items)) {
+    return false;
+  }
+  let isEmpty = true;
+  items.forEach((item) => {
+    if (!isEmpty) return;
+
+    if (['text', 'number', 'date', 'checkbox'].includes(item.type)) {
+      if (answers[item.id] && answers[item.id] !== '') isEmpty = false;
+    }
+    if (['arrayText', 'arrayNumber', 'arrayDate'].includes(item.type)) {
+      if (answers[item.id] && Array.isArray(answers[item.id]) && answers[item.id].length > 0) {
+        (answers[item.id] as string[]).forEach(a => { if (a && a !== '') isEmpty = false; })
+      }
+    }
+  });
+  return !isEmpty;
+};
+
+/** Evaluates an answer value to a boolean. False if the value is empty or false, otherwise true. */
+export const evaluateAnswer = (
+  questionId: string,
+  answers: Record<string, any>,
+  allQuestions: Question[]
+): boolean => {
+  const question = allQuestions.find((q) => q.id === questionId);
+  if (!question) return false;
+
+  switch (question.type) {
+    case 'checkbox':
+      if (answers[questionId]) return answers[questionId];
+      return false;
+    case 'text':
+    case 'number':
+    case 'date':
+      return (answers[questionId] && answers[questionId] !== '');
+    case 'editableList':
+      return (
+        answers[questionId] &&
+        Object.keys(answers[questionId]).filter(
+          (key) => answers[questionId][key] && answers[questionId][key] !== ''
+        ).length > 0
+      );
+    case 'repeaterField':
+      return (answers[questionId] && answers[questionId].length > 0);
+    case 'summaryList':
+      return evaluateSummaryList(answers, question.items || []);
+    default:
+      return false;
+  }
```

Comment:
extract to functions. switches = bad

Reply:
Don't understand exactly what you mean here... how would you refactor this?

Reply:
Would be good if you could give an example of how to replace this switch and why it's bad in general!

Reply:
I guess we could rewrite this switch statement to something along the lines of

if (question.type === 'checkbox') {
 ...
}

if ([''text', 'number', 'date'].includes(question.type)) {
  ...
}

etc

But again @dannil76 i would like to know the benefit of not using a switch. In other words what do you se as a danger in using switch case?

Reply:
Kan vara problematsikt att få switch att göra endast EN sak. I sin natur gör alltid switch fler saker. Polymorphism är ett allternativ att kika på.

Reply:
@dannil76 Still not very clear what you mean, or how polymorphism can be used in this context? Can you give a concrete example of how you would refactor this?

Like, of course we can do it as Teddy suggests, with if-statements, but that feels functionally equivalent, and I don't really understand why it should be better.

Reply:
Teddys förslag ät tydligare att läsa.

Reply:
I rewrote it as a number of if-statements, but honestly I don't find it more readable myself. Switches were literally added in order to avoid this kind of long list of if-statements, but whatever...